mittab/apps/tab/outround_pairing_views.py

```diff
@@ -0,0 +1,639 @@
+        type_of_round=type_of_round,
+        num_teams=num_teams,
+        opp_team=t
+    ).exists()]
+
+    excluded_judges = Judge.objects.exclude(
+        judges_outrounds__num_teams=num_teams,
+        judges_outrounds__type_of_round=type_of_round,
+    ).exclude(
+        judges_outrounds__type_of_round=other_round_type,
+        judges_outrounds__num_teams=other_round_num
+    ).filter(
+        checkin__round_number=0
+    )
+
+    non_checkins = Judge.objects.exclude(
+        judges_outrounds__num_teams=num_teams,
+        judges_outrounds__type_of_round=type_of_round
+    ).exclude(
+        judges_outrounds__type_of_round=other_round_type,
+        judges_outrounds__num_teams=other_round_num
+    ).exclude(
+        checkin__round_number=0
+    )
+
+    available_rooms = Room.objects.exclude(
+        rooms_outrounds__num_teams=num_teams,
+        rooms_outrounds__type_of_round=type_of_round
+    ).exclude(
+        rooms_outrounds__num_teams=other_round_num,
+        rooms_outrounds__type_of_round=other_round_type
+    )
+
+    checked_in_rooms = [r.room for r in RoomCheckIn.objects.filter(round_number=0)]
+    available_rooms = [r for r in available_rooms if r in checked_in_rooms]
+
+    size = max(list(
+        map(
+            len,
+            [excluded_teams, excluded_judges, non_checkins, available_rooms]
+        )))
+    # The minimum rank you want to warn on
+    warning = 5
+    excluded_people = list(
+        zip(*[
+            x + [""] * (size - len(x)) for x in [
+                list(excluded_teams),
+                list(excluded_judges),
+                list(non_checkins),
+                list(available_rooms)
+            ]
+        ]))
+
+    return render(request,
+                  "outrounds/pairing_base.html",
+                  locals())
+
+
+def alternative_judges(request, round_id, judge_id=None):
+    round_obj = Outround.objects.get(id=int(round_id))
+    round_gov, round_opp = round_obj.gov_team, round_obj.opp_team
+    # All of these variables are for the convenience of the template
+    try:
+        current_judge_id = int(judge_id)
+        current_judge_obj = Judge.objects.get(id=current_judge_id)
+        current_judge_name = current_judge_obj.name
+        current_judge_rank = current_judge_obj.rank
+    except TypeError:
+        current_judge_id, current_judge_obj, current_judge_rank = "", "", ""
+        current_judge_name = "No judge"
+
+    var_to_nov = TabSettings.get("var_to_nov", 2)
+    other_round_num = round_obj.num_teams / var_to_nov
+    if round_obj.type_of_round == BreakingTeam.NOVICE:
+        other_round_num = round_obj.num_teams * var_to_nov
+
+    other_round_type = BreakingTeam.NOVICE \
+                       if round_obj.type_of_round == BreakingTeam.VARSITY \
+                          else BreakingTeam.VARSITY
+
+    excluded_judges = Judge.objects.exclude(
+        judges_outrounds__num_teams=round_obj.num_teams,
```

Comment:
I think it might be better to not have any logic sticking judges into a particular break, since sometimes tournaments will have novices judge novice semis, but have varsity judge finals, etc. I think the benefit we get here might not be worth the trade-off

Reply:
This logic doesn't stick any judges in a certain break, it just ensures that judges who are excluded are really excluded (ie not judges a novice round happening at the same time, etc).  This is just meant to allow a TD to use the interface a little bit faster, all checked in outround judges should show up, however, in the included_judges variable if they're paired in to either the varsity, or novice round.